examples/custom/src/lib.rs

```diff
@@ -38,55 +39,68 @@ mod code {
 #[cfg(feature = "std")]
 pub use code::WASM_BINARY_OPT as WASM_BINARY;
 
-#[derive(Decode, Encode)]
+#[derive(Debug, Decode, Encode)]
 pub enum InitMessage {
     Capacitor(String),
     BTree,
     BackendError,
+    Ping,
+    Vec,
+    PiggyBank,
+    Distributor,
+}
+
+pub trait Program {
+    fn init(args: gstd::Box<dyn gstd::any::Any>) -> Self
+    where
+        Self: Sized;
+
+    fn handle(&'static mut self) {}
+    fn state(&self) {}
 }
```

Comment:
May be I don't understand something but can we simplify this code like this for example? If init does not take something then we can just put `P = ()` there. 
```suggestion
pub trait Program<P>: Sized {
    fn init(init_payload: P) -> Self;
    fn handle(&mut self) {}
    fn state(&self) {}
}
```